milestone-3/front-end/src/App.js

```diff
@@ -0,0 +1,47 @@
+import './App.css';
+import {
+  BrowserRouter as Router,
+  Routes,
+  Route,
+  useLocation
+} from "react-router-dom";
+import recipes from './recipeData';
+import Navbar from './components/navbar';
+import About from './components/about';
+import Home from './components/home';
+import RecipePage from './components/recipePage';
+import { useLayoutEffect } from 'react'
+
+const Wrapper = ({children}) => {
+  const location = useLocation();
+  useLayoutEffect(() => {
+    document.documentElement.scrollTo(0, 0);
+  }, [location.pathname]);
+  return children
+} 
+
+function App() {
+  return (
+    <Router>
+      <Wrapper>
+        <div className="App">
+          <Navbar />
+
+          <Routes>
+            { recipes.map(recipe => (
+              <Route path={recipe.recipeRoute} element={<RecipePage {...recipe} />} />
+            ))}
```

Comment:
Nice! I never realized you could spread props into a component like that.

Reply:
Yeah! I figured there must be a way to do something like this... some digging through the router docs uncovered this.